driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -25,86 +25,483 @@
+
+    /**
+     * Returns the {@linkplain DocumentExpression document} value of the field
+     * with the provided {@code fieldName},
+     * or the {@code other} value
+     * if the document {@linkplain #has} no such field,
+     * or if the specified field is not a (child) document
+     * (or other {@linkplain Expression#isDocumentOr document-like value}.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     DocumentExpression getDocument(String fieldName, DocumentExpression other);
 
+    /**
+     * Returns the {@linkplain DocumentExpression document} value of the field
+     * with the provided {@code fieldName},
+     * or the {@code other} value
+     * if the document {@linkplain #has} no such field,
+     * or if the specified field is not a (child) document
+     * (or other {@linkplain Expression#isDocumentOr document-like value}.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     default DocumentExpression getDocument(final String fieldName, final Bson other) {
         return getDocument(fieldName, of(other));
     }
 
-    <T extends Expression> MapExpression<T> getMap(String fieldName);
+    /**
+     * Returns the {@linkplain MapExpression map} value of the field
+     * with the provided {@code fieldName}.
+     *
+     * <p>Warning: The type and presence of the resulting value is not
+     * enforced by the API. The use of this method is an
+     * unchecked assertion that the document
+     * {@linkplain #has(String) has} the named field and
+     * the field value is of the specified type.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @return the resulting value.
+     * @param <T> the type.
+     */
+    @MqlUnchecked({PRESENT, TYPE})
+    <T extends Expression> MapExpression<@MqlUnchecked(TYPE_ARGUMENT) T> getMap(String fieldName);
+
+
+    /**
+     * Returns the {@linkplain MapExpression map} value of the field
+     * with the provided {@code fieldName},
+     * or the {@code other} value
+     * if the document {@linkplain #has} no such field,
+     * or if the specified field is not a map
+     * (or other {@linkplain Expression#isMapOr} map-like value}).
+     *
+     * <p>Warning: The type argument of the resulting value is not
+     * enforced by the API. The use of this method is an
+     * unchecked assertion that the type argument is correct.
+     *
+     * @mongodb.server.release 5.0
+     * @param fieldName the name of the field.
+     * @param other the other value.
+     * @return the resulting value.
+     * @param <T> the type.
+     */
+    @MqlUnchecked({TYPE_ARGUMENT})
     <T extends Expression> MapExpression<T> getMap(String fieldName, MapExpression<? extends T> other);
```

Comment:
The placement of `MqlUnchecked` with `TYPE_ARGUMENT` is inconsistent within `DocumentExpression`.